src/planner/join_order_enumerator.cpp

```diff
@@ -483,8 +488,11 @@ void JoinOrderEnumerator::appendNonRecursiveExtend(std::shared_ptr<NodeExpressio
 
 void JoinOrderEnumerator::appendRecursiveExtend(std::shared_ptr<NodeExpression> boundNode,
     std::shared_ptr<NodeExpression> nbrNode, std::shared_ptr<RelExpression> rel,
-    common::RelDirection direction, LogicalPlan& plan) {
+    common::ExtendDirection direction, LogicalPlan& plan) {
     auto hasAtMostOneNbr = extendHasAtMostOneNbrGuarantee(*rel, *boundNode, direction, catalog);
+    if (!rel->isDirected()) {
```

Comment:
you can assert !isDirected here. We will add support to recursive join progressively.